File: app/src/main/java/org/nitri/opentopo/nearby/viewmodel/NearbyViewModel.java
package org.nitri.opentopo.nearby.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import org.nitri.opentopo.nearby.entity.NearbyItem;
import org.nitri.opentopo.nearby.repo.NearbyRepository;

import java.util.List;

/**
 * Created by helfrich on 25/02/2018.
 */

public class NearbyViewModel extends ViewModel {

    private NearbyRepository mRepository;

    public void setRepository(NearbyRepository repository) {
        mRepository = repository;
    }

    public LiveData<List<NearbyItem>> getItems() {
        return mRepository.loadNearbyItems();
    }

}

File: app/src/main/java/org/nitri/opentopo/nearby/viewmodel/NearbyViewModel.kt
package org.nitri.opentopo.nearby.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.flow.*
import kotlinx.coroutines.launch
import org.nitri.opentopo.nearby.entity.NearbyItem
import org.nitri.opentopo.nearby.repo.NearbyRepository

class NearbyViewModel(private val mRepository: NearbyRepository) : ViewModel() {
     private val _items = MutableLiveData<List<NearbyItem>>()
    val items: LiveData<List<NearbyItem>> get() = _items

    init {
        viewModelScope.launch {
            mRepository.items.collect {
                _items.value = it
            }
        }
        mRepository.refresh(viewModelScope)
    }
}
